### PR TITLE
Fix proper permissions for multiple file access

### LINKED ISSUE #8890: getById returns incorrect permissions
1. Create a folder 'foo'
2. Create a file 'bar' in 'foo'
3. Share 'bar' to 'user2' read only
4. Share 'foo' to 'user2' read-write
5. Now request via the folder 'getById'

See that both returned entries have the same (to limited) permissions.

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #8890

In case you have access to a file via multiple ways, for example:
1. the file is shared with you with permission read only
2. the folder containing the file is shared with your read/write

Requesting the getById function on the userFolder would give back two
entries but both with the same permissions. Depending on the node you
picked this is not right.

@icewind1991 please have a look. I realize this is a little less optimized. But it is now correct I think.

### CODE REVIEW

File: lib/private/Files/Node/Folder.php [hunk 1]
```diff
@@ -323,6 +320,10 @@ public function getById($id) {
 			));
 		}, $mountsContainingFile);
 
+		$nodes = array_filter($nodes, function ($x) {
```
Comment (icewind1991) [inline lib/private/Files/Node/Folder.php]: You can leave out the callback argument to filter out any `falseish` values

**Review by icewind1991 — DISMISSED** [review]

**Comment by icewind1991** [comment]
Looks good otherwise

File: lib/private/Files/Node/Folder.php (hunk 1 above)
Comment (rullzer) [reply lib/private/Files/Node/Folder.php]: Ah fair enough. was late yesterday.

**Comment by rullzer** [comment]
I'll fix the tests later as well

**Comment by rullzer** [comment]
Review time!

**Review by ChristophWurst — APPROVED** [review]
Change makes sense and code looks good :+1: from me

**Review by icewind1991 — APPROVED** [review]

**Pull request merged**